xma/layers/linear_attention/module.py

```
@@ -36,26 +39,44 @@ def __init__(
         *,
         kernel_backend: KernelBackend | None = None,
-    ) -> tuple[torch.Tensor, torch.Tensor]:
+    ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor | None]:
         input = self.input_projection(input)
 
-        query, key, value = input.split((self.query_size, self.key_size, self.value_size), dim=-1)
+        if self.conv1d is None:
+            assert conv_state is None
+        else:
+            assert cu_seqlens is None, "depthwise causal conv does not support cu_seqlens"
+            input, conv_state = self.conv1d(input, input_state=conv_state, output_state=output_conv_state)
```

Ruff (0.16.0)

[error] Variable `input` is shadowing a Python builtin

(A001)